# Panel de Visualización de datos AirBnB Sidney

## Herramienta para el análisis, visualización y predicción de datos de alquileres de piso para uso vacacional en la ciudad de Sidney (Australia).

### 1. Importar librerías a utilizar en el análisis de datos y visualización

In [3]:
import pandas as pan
import random
import numpy as np
import seaborn as sns
import matplotlib
import matplotlib.pyplot as plt
import itertools as it
    

### 2. Importar base de datos .CSV

In [4]:
pan.read_csv ('sydney_airbnb.csv')

id                                               name    host_id  \
0         12351                  Sydney City & Harbour at the door      17061   
1         14250                                Manly Harbour House      55948   
2         15253  Stunning Penthouse Apartment In Heart Of The City      59850   
3         20865                 3 BED HOUSE + 1 BED STUDIO Balmain      64282   
4         26174                 COZY PRIVATE ROOM, GREAT LOCATION!     110561   
...         ...                                                ...        ...   
36657  30592081                             The top floor paradise  229475098   
36658  30592161   Sydney harbour catamaran sailing. Parties for 10  223580521   
36659  30592248  Double Room built-in a large and bright apartment  119396721   
36660  30592505     Bright Modern apartment in a Premiere location  229479027   
36661  30593866   Huge Sunny Double Room with Spa in Best Location   51138070   

             host_name  neighbourhood_group neighbourhood   latitude  \
0               Stuart                  NaN        Sydney -33.865153   
1                Heidi                  NaN         Manly -33.800929   
2                Morag                  NaN        Sydney -33.880455   
3                Fiona                  NaN    Leichhardt -33.859072   
4               Amanda                  NaN     Woollahra -33.889087   
...                ...                  ...           ...        ...   
36657          Sharney                  NaN     Woollahra -33.874940   
36658            Blake                  NaN    Canada Bay -33.847841   
36659           Nelson                  NaN   Strathfield -33.863719   
36660   Sofei & Kelvin                  NaN      Rockdale -33.929960   
36661  Aaron & Natalie                  NaN        Sydney -33.906938   

        longitude        room_type  price  minimum_nights  number_of_reviews  \
0      151.191896     Private room    100               2                493   
1      151.261722  Entire home/apt    471               5                  1   
2      151.216541     Private room    109               2                300   
3      151.172753  Entire home/apt    450               7                 15   
4      151.259404     Private room     62               1                 42   
...           ...              ...    ...             ...                ...   
36657  151.259375  Entire home/apt    190              10                  0   
36658  151.149573  Entire home/apt    900               1                  0   
36659  151.081721     Private room     46               4                  0   
36660  151.150659  Entire home/apt    170               2                  0   
36661  151.211255     Private room     80               5                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
0      2018-11-21               4.83                               2   
1      2016-01-02               0.03                               2   
2      2018-11-16               3.63                               2   
3      2018-01-07               0.18                               1   
4      2018-11-23               0.45                               1   
...           ...                ...                             ...   
36657         NaN                NaN                               1   
36658         NaN                NaN                               1   
36659         NaN                NaN                               1   
36660         NaN                NaN                               1   
36661         NaN                NaN                               3   

       availability_365  
0                   187  
1                   321  
2                   316  
3                    69  
4                   140  
...                 ...  
36657               172  
36658               168  
36659                11  
36660                89  
36661                11  

[36662 rows x 16 columns]

In [6]:
df = pan.read_csv ('sydney_airbnb.csv')
df

id                                               name    host_id  \
0         12351                  Sydney City & Harbour at the door      17061   
1         14250                                Manly Harbour House      55948   
2         15253  Stunning Penthouse Apartment In Heart Of The City      59850   
3         20865                 3 BED HOUSE + 1 BED STUDIO Balmain      64282   
4         26174                 COZY PRIVATE ROOM, GREAT LOCATION!     110561   
...         ...                                                ...        ...   
36657  30592081                             The top floor paradise  229475098   
36658  30592161   Sydney harbour catamaran sailing. Parties for 10  223580521   
36659  30592248  Double Room built-in a large and bright apartment  119396721   
36660  30592505     Bright Modern apartment in a Premiere location  229479027   
36661  30593866   Huge Sunny Double Room with Spa in Best Location   51138070   

             host_name  neighbourhood_group neighbourhood   latitude  \
0               Stuart                  NaN        Sydney -33.865153   
1                Heidi                  NaN         Manly -33.800929   
2                Morag                  NaN        Sydney -33.880455   
3                Fiona                  NaN    Leichhardt -33.859072   
4               Amanda                  NaN     Woollahra -33.889087   
...                ...                  ...           ...        ...   
36657          Sharney                  NaN     Woollahra -33.874940   
36658            Blake                  NaN    Canada Bay -33.847841   
36659           Nelson                  NaN   Strathfield -33.863719   
36660   Sofei & Kelvin                  NaN      Rockdale -33.929960   
36661  Aaron & Natalie                  NaN        Sydney -33.906938   

        longitude        room_type  price  minimum_nights  number_of_reviews  \
0      151.191896     Private room    100               2                493   
1      151.261722  Entire home/apt    471               5                  1   
2      151.216541     Private room    109               2                300   
3      151.172753  Entire home/apt    450               7                 15   
4      151.259404     Private room     62               1                 42   
...           ...              ...    ...             ...                ...   
36657  151.259375  Entire home/apt    190              10                  0   
36658  151.149573  Entire home/apt    900               1                  0   
36659  151.081721     Private room     46               4                  0   
36660  151.150659  Entire home/apt    170               2                  0   
36661  151.211255     Private room     80               5                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
0      2018-11-21               4.83                               2   
1      2016-01-02               0.03                               2   
2      2018-11-16               3.63                               2   
3      2018-01-07               0.18                               1   
4      2018-11-23               0.45                               1   
...           ...                ...                             ...   
36657         NaN                NaN                               1   
36658         NaN                NaN                               1   
36659         NaN                NaN                               1   
36660         NaN                NaN                               1   
36661         NaN                NaN                               3   

       availability_365  
0                   187  
1                   321  
2                   316  
3                    69  
4                   140  
...                 ...  
36657               172  
36658               168  
36659                11  
36660                89  
36661                11  

[36662 rows x 16 columns]

### 3.  Asignar la base de datos a un DataFrame (df)

### 4. Exploración inicial de datos

> #### 4.1. Tipo de objetos (int,string, object, boolean, range).

> #### 4.2. Nombre y tipo de variables (columnas).

> #### 4.3. Información de filas (primeras y últimas).

> #### 4.4. Datos nulos (conteo e identificación).

> #### 4.5. Datos duplicados (conteo e identificación).

### 5. Ingeniería de características

### 6. Visualización de datos

### 7. Análisis estadístico de datos

### Panel de Visualización de datos (link Streamlit)

### Tablero de trabajo (desarrollo del proyecto) : Link Trolle

> *Mapeo de ubicaciones de los inmuebles AirBnB a arrendar en Sidney*

**Ejemplo con bicicleterias de Buenos Aires Argentina**

dataset de bicicleterías de Ciudad de Buenos Aires
descargado desde https://data.buenosaires.gob.ar/dataset/bicicleterias

bici = pd.read_csv('data/bicicleterias.csv', sep=';')
bici.head()

corregimos el campo de coordenadas del dataset.

def coord(c):
    coor = re.findall(r'-?\d+\.\d{7}', c)
    coords = [float(s) for s in coor]
    return coords[::-1]

bici['WKT'] = bici['WKT'].apply(coord)

filtramos solo las bicicleterías de palermo

bici_palermo = bici[bici.BARRIO == 'PALERMO'][['WKT', 'NOMBRE']]

creamos el mapa con folium

mapa = folium.Map(location=[-34.588889, -58.430556], zoom_start=13)

agregamos los markers con el nombre de cada bicicletería.

for index, row in bici_palermo.iterrows():
    mapa.simple_marker(row['WKT'], 
                   popup=row['NOMBRE'], marker_color='red',
                   marker_icon='info-sign')
    
    
visualizamos el mapa con los markers

mapa




In [8]:
# Ejemplos básicos de Follium

import folium

m = folium.Map(location=[40.965, -5.664])
m

In [9]:
folium.Map(
    location=[40.965, -5.664],
    tiles='Stamen Toner',
    zoom_start=16
)

In [10]:
m = folium.Map(
    location=[40.965, -5.664],
    zoom_start=12,
    tiles='Stamen Terrain'
)

tooltip = 'plaza Mayor'

folium.Marker([40.965, -5.664], popup='Plaza Mayor', tooltip=tooltip).add_to(m)


m

# Mapa interactivo de Sydney Australia (follium)

**Información de Sydney Australia**

*Coordenadas geográficas de Sídney* 	

Latitud: -33.8667, 

Longitud: 151.2 33° 52′ 0″ Sur, 
          151° 12′ 0″ Este

In [12]:
import folium

m = folium.Map(location=[-33.8667,151.2])
m

  **Stamen Toner (tesela raster)**

In [13]:
folium.Map(
    location=[-33.86785,151.20732],
    tiles='Stamen Toner',
    zoom_start=16
)

**Stamen Terrain (tesela raster)**

In [15]:
m = folium.Map(
    location=[-33.86785,151.20732],
    zoom_start=12,
    tiles='Stamen Terrain'
)

tooltip = 'Ópera de Sidney'

folium.Marker([-33.8567844, 151.2152967], popup='Ópera de Sidney', tooltip=tooltip).add_to(m)


m


#Ópera de Sidney

#Latitud: -33.8567844. Longitud: 151.2152967 